crates/biome_css_parser/src/syntax/css_dimension.rs

```diff
@@ -0,0 +1,110 @@
+use crate::parser::CssParser;
+use biome_css_syntax::CssSyntaxKind::*;
+use biome_css_syntax::T;
+use biome_parser::prelude::ParsedSyntax;
+use biome_parser::prelude::ParsedSyntax::{Absent, Present};
+use biome_parser::Parser;
+
+use super::parse_error::expected_unit;
+use super::{parse_regular_identifier, parse_regular_number};
+
+#[inline]
+pub(crate) fn is_at_css_dimension(p: &mut CssParser) -> bool {
+    is_at_css_percentage_dimension(p) || is_at_css_regular_dimension(p)
+}
+
+#[inline]
+pub(crate) fn parse_css_dimension(p: &mut CssParser) -> ParsedSyntax {
+    if !is_at_css_dimension(p) {
+        return Absent;
+    }
+    if is_at_css_percentage_dimension(p) {
+        return parse_css_percentage_dimension(p);
+    }
+    parse_css_regular_dimension(p)
+}
+
+fn is_at_css_percentage_dimension(p: &mut CssParser) -> bool {
+    p.at(CSS_NUMBER_LITERAL) && matches!(p.nth(1), T![%])
+}
+#[inline]
+fn parse_css_percentage_dimension(p: &mut CssParser) -> ParsedSyntax {
+    if !is_at_css_percentage_dimension(p) {
+        return Absent;
+    }
+    let m = p.start();
+    let _css_number = parse_regular_number(p);
+    p.expect(T![%]);
+    Present(m.complete(p, CSS_PERCENTAGE))
+}
+fn is_at_css_regular_dimension(p: &mut CssParser) -> bool {
+    p.at(CSS_NUMBER_LITERAL) && p.nth_at(1, T![ident])
+}
+#[inline]
+fn parse_css_regular_dimension(p: &mut CssParser) -> ParsedSyntax {
+    if !is_at_css_regular_dimension(p) {
+        return Absent;
+    }
+    let m = p.start();
+    let _css_number = parse_regular_number(p);
+    parse_css_unit(p).or_add_diagnostic(p, expected_unit);
+    Present(m.complete(p, CSS_REGULAR_DIMENSION))
+}
+
+#[inline]
+fn parse_css_unit(p: &mut CssParser) -> ParsedSyntax {
+    if !(p.at(T![ident]) && is_unit_str(p.cur_text())) {
+        return Absent;
+    }
+    parse_regular_identifier(p)
+}
+
+#[inline]
+fn is_unit_str(unit: &str) -> bool {
+    is_length_unit(unit)
```

Comment:
I'm wondering if we need to have "em", "rem" as keywords. 
See 
https://github.com/biomejs/biome/blob/6739a5148b41e3d81b19722fffe17d97de4434dc/xtask/codegen/src/css_kinds_src.rs#L47
https://github.com/biomejs/biome/blob/6739a5148b41e3d81b19722fffe17d97de4434dc/xtask/codegen/css.ungram#L223
https://github.com/biomejs/biome/blob/6739a5148b41e3d81b19722fffe17d97de4434dc/crates/biome_css_parser/src/lexer/mod.rs#L727-L770

We can move them to keywords in another PR.

Reply:
I tried but got the following error, we can fix it in the next pr.
```
-> biome\crates\biome_parser\src\token_set.rs:46:18
   |
46 |         _ => [0, 1u128 << (num - 127)],
   |                  ^^^^^^^^^^^^^^^^^^^^ attempt to shift left by `170_usize`, which would overflow
```

We may need a modification such as the following
```rust
--- a/crates/biome_parser/src/token_set.rs
+++ b/crates/biome_parser/src/token_set.rs
@@ -2,10 +2,10 @@ use biome_rowan::SyntaxKind;
 use std::marker::PhantomData;

 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct TokenSet<K: SyntaxKind>([u128; 2], PhantomData<K>);
+pub struct TokenSet<K: SyntaxKind>([u128; 3], PhantomData<K>);

 impl<K: SyntaxKind> TokenSet<K> {
-    pub const EMPTY: TokenSet<K> = TokenSet([0; 2], PhantomData);
+    pub const EMPTY: TokenSet<K> = TokenSet([0; 3], PhantomData);

     pub fn singleton(kind: K) -> Self {
         unsafe { TokenSet::from_raw(kind.to_raw().0) }
@@ -13,7 +13,7 @@ impl<K: SyntaxKind> TokenSet<K> {

     pub const fn union(self, other: TokenSet<K>) -> Self {
         TokenSet(
-            [self.0[0] | other.0[0], self.0[1] | other.0[1]],
+            [self.0[0] | other.0[0], self.0[1] | other.0[1],self.0[2] | other.0[2]],
             PhantomData,
         )
     }
@@ -23,7 +23,8 @@ impl<K: SyntaxKind> TokenSet<K> {
         let num = kind as usize;
         match num {
             0..=127 => self.0[0] & mask(kind)[0] != 0,
-            _ => self.0[1] & mask(kind)[1] != 0,
+            128..=255 => self.0[1] & mask(kind)[1] != 0,
+            _ => self.0[2] & mask(kind)[2] != 0,
         }
     }

@@ -39,11 +40,12 @@ impl<K: SyntaxKind> TokenSet<K> {
     }
 }

-const fn mask(kind: u16) -> [u128; 2] {
+const fn mask(kind: u16) -> [u128; 3] {
     let num = kind as usize;
     match num {
-        0..=127 => [1u128 << num, 0],
-        _ => [0, 1u128 << (num - 127)],
+        0..=127 => [1u128 << num, 0,0],
+        128..=255 => [0,1u128 << (num - 127), 0],
+        _ => [0,0, 1u128 << (num - 255)],
     }
 }

```

Reply:
Oh, I see.. 
Maybe we can remove all colours from keywords for now.
Anyway we can do it in another PR.